File: snippets/rust/src/send_payment.rs
use std::sync::Arc;

use anyhow::Result;
use breez_sdk_core::*;

async fn send_payment(sdk: Arc<BreezServices>) -> Result<()> {
    // ANCHOR: send-payment
    let use_trampoline = false;
    // The `amount_msat` param is optional and should only passed if the bolt11 doesn't specify an amount.
    // The amount_msat is required in case an amount is not specified in the bolt11 invoice'.
    let optional_amount_msat: Option<u64> = None;
    let optional_label = Some("<label>".to_string());

    let req = SendPaymentRequest {
        bolt11: "...".into(),
        use_trampoline,
        amount_msat: optional_amount_msat,
        label: optional_label,
    };
    let response = sdk.send_payment(req).await?;
    // ANCHOR_END: send-payment

    Ok(())
}

File: snippets/go/send_payment.go
package example

import (
	"log"

	"github.com/breez/breez-sdk-go/breez_sdk"
)

func SendPayment() error {
	// ANCHOR: send-payment
	bolt11 := "bolt11 invoice"
	useTrampoline := true
	// The `amountMsat` param is optional and should only passed if the bolt11 doesn't specify an amount.
	// The amountMsat is required in case an amount is not specified in the bolt11 invoice'.
	optionalAmountMsat := uint64(3_000_000)
	optionalLabel := "<label>"
	sendPaymentRequest := breez_sdk.SendPaymentRequest{
		Bolt11:        bolt11,
		UseTrampoline: useTrampoline,
		AmountMsat:    &optionalAmountMsat,
		Label:         &optionalLabel,
	}
	response, err := sdk.SendPayment(sendPaymentRequest)
	if err != nil {
		return err
	}
	log.Printf("%#v", response)
	// ANCHOR_END: send-payment
	return nil
}
